File: go/hamming/hamming.go
package hamming

import "errors"

const testVersion = 5

// Distance calculates the hamming distance for two strands of DNA
func Distance(a, b string) (d int, err error) {
	if len(b) != len(a) {
		return 0, errors.New("Provided DNA strands have different lengths")
	}

	for i := 0; i < len(a); i++ {
		if a[i] != b[i] {
			d++
		}
	}

	return
}

File: ecmascript/hamming/hamming.js
class Hamming {
  compute(a, b) {
    if (a.length !== b.length) {
      throw new Error("DNA strands must be of equal length.");
    }

    let distance = 0;
    for (var i = 0; i < a.length; i++) {
      if (a[i] !== b[i]) {
        distance++;
      }
    }

    return distance;
  }
}

export default Hamming;
